### PR TITLE
autofarm code cleanup

### PR DESCRIPTION
**Pull request by ab9rf**
This is a minor cleanup of autofarm, mostly to eliminate some code smells (like `using std`) and to be more consistent with modern C++ coding standards

### CODE REVIEW

File: plugins/autofarm.cpp [hunk 1]
```diff
@@ -168,41 +162,37 @@ class AutoFarm {
...
+        const uint32_t bad_flags{
+#define F(x) (df::item_flags::Mask::mask_##x)
+        F(dump) | F(forbid) | F(garbage_collect) |
+        F(hostile) | F(on_fire) | F(rotten) | F(trader) |
+        F(in_building) | F(construction) | F(artifact)
 #undef F
+        };
 
-        for (auto ii : world->items.other[df::items_other_id::SEEDS])
+        for (auto& ii : world->items.other[df::items_other_id::SEEDS])
```
Comment (myk002) [inline plugins/autofarm.cpp:175]: Is there a benefit to using a reference to a pointer here?

File: plugins/autofarm.cpp [hunk 2]
```diff
@@ -213,11 +203,11 @@ class AutoFarm {
             farm->plant_id[season] = new_plant_id;
             out << "autofarm: changing farm #" << farm->id <<
                 " from " << get_plant_name(old_plant_id) <<
-                " to " << get_plant_name(new_plant_id) << endl;
+                " to " << get_plant_name(new_plant_id) << '\n';
```
Comment (myk002) [inline plugins/autofarm.cpp:206]: I think it's functionally equivalent, but isn't the standard to use `std::endl` in this context?

File: plugins/autofarm.cpp [hunk 3]
```diff
@@ -213,11 +203,11 @@ class AutoFarm {
...
         }
     }
 
-    void set_farms(color_ostream& out, set<int> plants, vector<df::building_farmplotst*> farms)
+    void set_farms(color_ostream& out, std::set<int>& plants, std::vector<df::building_farmplotst*>& farms)
```
Comment (myk002) [inline plugins/autofarm.cpp:210]: plants could be made const. I think farms can too. That way the change to passing by reference can guarantee no side effects.

File: plugins/autofarm.cpp [hunk 4]
```diff
@@ -281,44 +268,33 @@ class AutoFarm {
...
-            if (i &&
-                (i->flags.whole & bad_flags.whole) == 0 &&
-                plantable_plants.count(i->mat_index) > 0)
-            {
-                lastCounts[i->mat_index] += i->stack_size;
-            }
-        }
+        };
 
+        for (auto& i : world->items.other[df::items_other_id::PLANT])
```
Comment (myk002) [inline plugins/autofarm.cpp:290]: Elegant!

File: plugins/autofarm.cpp [hunk 5]
```diff
@@ -281,44 +268,33 @@ class AutoFarm {
...
+            count(i);
+        for (auto& i : world->items.other[df::items_other_id::PLANT_GROWTH])
```
Comment (myk002) [inline plugins/autofarm.cpp:292]: This is a pointer type, right? No need for the &, I think

Comment (ab9rf) [reply plugins/autofarm.cpp:292]: but also harmless

File: plugins/autofarm.cpp (hunk 2 above)
Comment (ab9rf) [reply plugins/autofarm.cpp:206]: `endl` forces an output flush, there's no reason to do that here

File: plugins/autofarm.cpp (hunk 4 above)
Comment (ab9rf) [reply plugins/autofarm.cpp:290]: it annoyed me that i had repeated code. i'm a big fan of DRY

File: plugins/autofarm.cpp (hunk 1 above)
Comment (ab9rf) [reply plugins/autofarm.cpp:175]: not really, but at the same time there's no real cost to it

Comment (myk002) [reply plugins/autofarm.cpp:175]: Fair enough

**Review by myk002 — APPROVED** [review]

**Pull request merged**